<launch>
    <!-- This launch file starts both the original DOGM node and the new GAT+Transformer node. -->

    <!-- Arguments -->
    <arg name="use_radar" default="true" />
    <arg name="show_viz" default="true" />
    <arg name="model_path" default="$(find dogm_ros)/trained_models/gat_transformer.pth" /> <!-- TODO: Update this path to your trained model -->

    <!-- GAT+Transformer Python Node -->
    <node pkg="dogm_ros" type="gat_transformer_node.py" name="gat_transformer_node" output="screen">
        <!-- Remap topics if they are different from the defaults in the script -->
        <!-- 
        <remap from="/lidar/points" to="/your_lidar_topic" />
        <remap from="/radar/points" to="/your_radar_topic" />
        <remap from="/odom" to="/your_odom_topic" />
        -->
        <param name="model_path" value="$(arg model_path)" />
    </node>

    <!-- Original DOGM C++ Node -->
    <!-- It will automatically subscribe to the /filtered_sigma topic published by the python node -->
    <node pkg="dogm_ros" type="dogm_node" name="dogm_node" output="screen">
        <rosparam command="load" file="$(find dogm_ros)/config/params.yaml" />
        <param name="use_radar" value="$(arg use_radar)" />
    </node>

    <!-- Rviz for visualization -->
    <node if="$(arg show_viz)" pkg="rviz" type="rviz" name="rviz" args="-d $(find dogm_ros)/rviz/dogm.rviz" />

</launch>
